libyul/AST.h

```diff
@@ -101,6 +101,18 @@ struct Continue { langutil::DebugData::ConstPtr debugData; };
 /// Leave statement (valid within function)
 struct Leave { langutil::DebugData::ConstPtr debugData; };
 
+/// Immutable AST comprised of its top-level block
+class AST
+{
+public:
+	explicit AST(Block _block): m_block(std::move(_block)) {}
+
+	Block const& block() const { return m_block; }
```

Comment:
This should have a more descriptive name. Yes, it's a block but there are many blocks in the AST and that's not the most important thing about it :) It's specifically the top-level one that's the root of the tree.

```suggestion
	Block const& root() const { return m_root; }
```